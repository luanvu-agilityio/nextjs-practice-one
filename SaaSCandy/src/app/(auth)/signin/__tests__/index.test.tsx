import { render, screen } from '@testing-library/react';
import SignInPage from '@/app/(auth)/signin/page';

jest.mock('@/components/pages', () => ({
  SignInPageContent: () => <div data-testid='signin-form'>Sign In Form</div>,
}));

describe('SignInPage - Snapshot Tests', () => {
  it('should match snapshot', () => {
    const { container } = render(<SignInPage />);
    expect(container).toMatchSnapshot();
  });

  it('should render with correct wrapper classes', () => {
    const { container } = render(<SignInPage />);
    const wrapper = container.firstChild as HTMLElement;

    expect(wrapper).toHaveClass('py-16');
  });

  it('should render SignInPageContent component', () => {
    render(<SignInPage />);
    expect(screen.getByTestId('signin-form')).toBeInTheDocument();
  });
});
