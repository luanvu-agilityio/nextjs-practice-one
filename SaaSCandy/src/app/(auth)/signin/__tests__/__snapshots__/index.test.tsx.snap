// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SignInPage - Snapshot Tests should match snapshot 1`] = `
<div>
  <div
    class="py-16"
  >
    <div
      data-testid="signin-form"
    >
      Sign In Form
    </div>
  </div>
</div>
`;
